packages/legend-query/src/stores/QueryFunctionsExplorerState.ts

```diff
@@ -0,0 +1,456 @@
+      const childTreeNodeData = generateFunctionsExplorerTreeNodeData(
+        queryBuilderState,
+        childPackage,
+        rootPackageName,
+      );
+      addUniqueEntry(rootIds, childTreeNodeData.id);
+      nodes.set(childTreeNodeData.id, childTreeNodeData);
+    });
+  return { rootIds, nodes };
+};
+
+export const getFunctionsExplorerTreeNodeChildren = (
+  queryBuilderState: QueryBuilderState,
+  node: QueryBuilderFunctionsExplorerTreeNodeData,
+  data: TreeData<QueryBuilderFunctionsExplorerTreeNodeData>,
+  rootPackageName = ROOT_PACKAGE_NAME.MAIN,
+): QueryBuilderFunctionsExplorerTreeNodeData[] => {
+  generateFunctionsExplorerTreeNodeChilrdren(
+    queryBuilderState,
+    node,
+    data,
+    rootPackageName,
+  );
+  return node.childrenIds
+    .map((id) => data.nodes.get(id))
+    .filter(isNonNullable)
+    .sort((a, b) => a.label.localeCompare(b.label))
+    .sort(
+      (a, b) =>
+        (b.packageableElement instanceof Package ? 1 : 0) -
+        (a.packageableElement instanceof Package ? 1 : 0),
+    );
+};
+
+const getAllPackagesFromElement = (element: PackageableElement): Package[] => {
+  if (element.package) {
+    return [element.package].concat(getAllPackagesFromElement(element.package));
+  }
+  return [];
+};
+
+export const generateFunctionSignature = (
+  element: ConcreteFunctionDefinition,
+  fullPath: boolean,
+): string =>
+  `${fullPath ? element.path : element.name}(${element.parameters
+    .map((p) => `${p.name}: ${p.type.value.name}[${p.multiplicity.str}]`)
+    .join(', ')})` +
+  `: ${element.returnType.value.name}[${element.returnMultiplicity.str}]`;
+
+export const generateDefaultValueForPrimitiveParameter = (
+  type: Type | undefined,
+  index: number,
+): string | number | boolean => {
+  if (!type) {
+    return `param${index}`;
+  }
+  switch (type.name) {
+    case PRIMITIVE_TYPE.BOOLEAN:
+      return true;
+    case PRIMITIVE_TYPE.FLOAT:
+    case PRIMITIVE_TYPE.DECIMAL:
+      return 0.0;
+    case PRIMITIVE_TYPE.NUMBER:
+    case PRIMITIVE_TYPE.INTEGER:
+      return 0;
+    case PRIMITIVE_TYPE.DATE:
+    case PRIMITIVE_TYPE.STRICTDATE:
+      return `%${format(new Date(2020, 10, 1), DATE_FORMAT)}`;
+    case PRIMITIVE_TYPE.DATETIME:
+      return `%${format(new Date(2020, 10, 1), DATE_TIME_FORMAT)}`;
+    case PRIMITIVE_TYPE.STRICTTIME:
+      return `%00:00:00`;
+    case PRIMITIVE_TYPE.STRING:
+      return "''";
+    default:
+      return `param${index}`;
+  }
+};
+
+export const generateFunctionLambdaString = (
+  element: ConcreteFunctionDefinition,
+): string => {
+  let lambdaString = '';
+  if (element.parameters.length > 0) {
+    for (let i = 0; i < element.parameters.length; i++) {
+      const paramType = element.parameters.at(i)?.type.value;
+      const separator = i !== element.parameters.length - 1 ? ', ' : '';
+      if (paramType instanceof Enumeration) {
+        lambdaString = `${lambdaString + paramType.path}.${
+          paramType.values.at(0)?.name
```

Comment:
You haven't accounted for the case where the enumeration have no enum values here, I suggest we fall back to `param${index}`.